src/screens/ShareTrip/ShareTrip.tsx

```diff
@@ -23,26 +34,145 @@ type Props = {
 };
 
 export default function ShareTrip(props: Props): JSX.Element {
+    console.log(props.route);
     const { t } = useTranslation();
-    const trip = props.route.params.trip;
+    const [mode, setMode] = useState<Mode>(
+        props.route.params.tripId ? Mode.SHARE_TRIP : Mode.JOIN_TRIP
+    );
 
-    const [execute, { loading }] = useCreateInvitationMutation();
-    const getInvitationLinkPromise = (): Promise<FetchResult<CreateInvitationMutation>> => {
-        return execute({
-            variables: {
-                input: {
-                    tripId: trip.id,
+    useEffect(() => {
+        async function getInitialUrl(): Promise<void> {
+            const link = await Linking.getInitialURL();
+            if (link && link !== null) {
+                console.log("we got a initial URL link ded");
+                console.log(link);
+            }
+        }
+        getInitialUrl();
+        Linking.addEventListener("url", (event) => {
+            console.log("got event listener callback");
+            console.log(event);
```

Comment:
Maybe make this log output a bit more specific because I wouldn't be able to tell where it's coming from if I don't know to look here.